### PR TITLE
created payment controllers

### PR DESCRIPTION
**Pull request by MUGABA**
### What this PR does (required):
-add the payment through `post /payments`
- retrieves all payments for the logged in user through `get /payments`
- get a single payment details through `get /payments/:paymentId`
- cancel the payment through `put /payments/: paymentId/cancel` 
- make a payment through `post /payments/:paymentId/pay`

### CODE REVIEW

File: server/controllers/payment.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
+const User = require("../models/User");
+const Profile = require("../models/Profile");
+const asyncHandler = require("express-async-handler");
+const generateToken = require("../utils/generateToken");
+const stripe = require("stripe")(process.env.STRIPE_SECRET);
+
+// @route POST /payments/
+// @desc add payment
+// @access Public
```
Comment (ezeilo-su) [inline server/controllers/payment.js:9]: Good documentation :+1:

File: server/controllers/payment.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+exports.addPayment = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+  const { sitterId, rate, startTime, endTime } = req.body;
```
Comment (ezeilo-su) [inline server/controllers/payment.js:12]: Good security practice

File: server/controllers/payment.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  return res.status(200).send({
+    success: {
+      payment: createPayment,
+    },
+  });
+});
+
+// @route GET /payments/:paymentId
+// @desc fetch payment record by id
+// @access Private
```
Comment (ezeilo-su) [inline server/controllers/payment.js:44]: :+1:

File: server/controllers/payment.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  try {
+    await stripe.charges.create({
+      customer: stripeAccountId,
+      amount: totalPayment,
+      currency: "cad",
+      source: "Hatchways",
+      description: `Made a payment of ${totalPayment}`,
+    });
+  } catch (error) {
+    res.status(400);
```
Comment (ezeilo-su) [inline server/controllers/payment.js:84]: I'd suggest a `500` status for this since it's not a client error.

**Review by ezeilo-su — COMMENTED** [review]
Looks good overall!
Happy coding!

File: server/controllers/payment.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+exports.getPayment = asyncHandler(async (req, res, next) => {
+  const paymentId = req.params.paymentId;
+  const payment = await Payment.findById(paymentId);
+  if (!payment) {
+    res.status(404);
+    throw new Error("Payment not found");
+  }
+  return res.status(200).send({
+    success: {
+      payment: payment,
```
Comment (moffatethan) [inline server/controllers/payment.js:54]: We can just pass 

```
{
   success: {
      payment 
   }
}
```

Comment (moffatethan) [reply server/controllers/payment.js:54]: Ditto throughout the PR

File: server/controllers/payment.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  const { totalPayment, stripeAccountId } = payment;
+
+  try {
+    await stripe.charges.create({
+      customer: stripeAccountId,
+      amount: totalPayment,
+      currency: "cad",
+      source: "Hatchways",
```
Comment (moffatethan) [inline server/controllers/payment.js:80]: Let's not have the source be Hatchways - remember the app is called Loving Sitter 🙂

**Review by moffatethan — CHANGES_REQUESTED** [review]

**Comment by moffatethan** [comment]
@MUGABA Any ETA on these changes? 🙂

**Comment by moffatethan** [comment]
@MUGABA Checking in again - any ETA?

**Pull request closed without merging**